### PR TITLE
Fix PulsarConsumerTest

### PR DESCRIPTION
**Pull request by xiangfu0**
Another try for fixing the PulsarConsumerTest.

### CODE REVIEW

File: pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConsumerTest.java [hunk 1]
```diff
@@ -164,52 +165,70 @@ public void publishRecords()
...
+
+            if (!_partitionToFirstMessageIdMapBatch.containsKey(partition)) {
+              _partitionToFirstMessageIdMapBatch.put(partition, messageId);
+            }
+          });
+        }
+        producer.flush();
       }
-      producer.flush();
+      waitForCondition(input -> validatePartitionMessageCount(partition, NUM_RECORDS_PER_PARTITION), 5 * 1000L,
```
Comment (richardstartin) [inline pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConsumerTest.java:213]: waiting 5 seconds at a time feels excessive

Comment (xiangfu0) [reply pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConsumerTest.java:213]: just a different way to try :p

Comment (xiangfu0) [reply pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/test/java/org/apache/pinot/plugin/stream/pulsar/PulsarConsumerTest.java:213]: will change to 10 or 30 sec later

**Comment by xiangfu0** [comment]
Rerun 4 times, seems to be ok?

**Comment by KKcorps** [comment]
There's already ` waitForMessagesToPublish` in the code. I am not sure how this implantation will solve the error.

**Comment by KKcorps** [comment]
Also, it seems to fix itself in the previous PRs as well. I did try re-running it 100-200. times as well previously and it worked without any errors. Not sure what causes this once in a blue moon error.

**Review by KKcorps — APPROVED** [review]

**Comment by gortiz** [comment]
> Also, it seems to fix itself in the previous PRs as well. I did try re-running it 100-200. times as well previously and it worked without any errors. Not sure what causes this once in a blue moon error.
> 
> Approving it, since there's no harm in merging these changes.

Did you try that in your machine or in CI?

**Pull request merged**